airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/conditional_page_paginator.py

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, Optional
+
+import requests
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.interpolation.interpolated_boolean import InterpolatedBoolean
+from airbyte_cdk.sources.declarative.states.dict_state import DictState
+
+
+class ConditionalPagePaginator:
+    def __init__(self, stop_condition_template: str, state: DictState, decoder: Decoder, config, page_key: str = None):
+        self._page_key = page_key or "page"
```

Comment:
I haven't seen a case where a connector needs a configurable page key that isn't `page`. I left it in to discuss, but I'd favor removing configurability until we know we need it. What do you think?